specification/common/attribute-precedence.md

```diff
@@ -0,0 +1,199 @@
+  - [Considerations](#considerations)
+  - [Example](#example)
+  - [Useful links](#useful-links)
+
+<!-- tocstop -->
+
+</details>
+
+## Overview
+
+This document provides supplementary guidelines for the attribute precedence 
+that exporters should follow when translating from the hierarchical OTLP format
+to non-hierarchical formats.
+
+A mapping is required when flattening out attributes from the structured OTLP
+format has attributes at different levels (e.g., Resource attributes, 
+InstrumentationLibraryScope attributes, Attributes on Spans/Metrics/Logs) to a
+non-hierarchical representation (e.g., OpenMetrics labels).
+In the case of OpenMetrics, the set of labels is completely flat and must have 
+unique labels only 
+(https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#labelset).
+Since OpenTelemetry allows for different levels of attributes, it is feasible
+that the same attribute appears multiple times on different levels.
+
+This document aims to provide guidance around consistently transforming 
+OpenTelemetry attributes to flat sets.
+
+## Attribute hierarchy in the OTLP message
+
+Since the OTLP format is a hierarchical format, there is an inherent order in 
+the attributes.
+In this document, 
+[Resource](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md)
+attributes are considered to be at the top of the hierarchy, since they are the
+most general attributes. 
+Attributes on individual Spans/ Metric data points/Logs are at the bottom of the
+hierarchy, as they are most specialized and only apply to a subset of all data.
+
+**A more specialized attribute that shares an attribute key with more general 
+attribute will take precedence.** 
+
+Therefore, more specialized attributes will overwrite more general ones.
+In some cases it might be desirable to overwrite an attribute like this.
+<!-- TODO example -->
+
+When de-normalizing the structured OTLP message to a flat set of key-value 
+pairs, attributes that are present at Resource and InstrumentationLibraryScope 
+level will be duplicated for each Span/Metric data point/Log.
+
+## Precedence per Signal
+
+Below, the precedence for each of the signals is spelled out explicitly.
+Some attributes (e.g., Span Link attributes) should only be used when flattening
+out attributes to transform the respective concept (Span Link attributes should
+not be added when transforming attributes on spans, for example).
+
+`A > B` denotes that the attribute on `A` will overwrite the attribute on  `B`
+if the keys clash.
+
+### Traces
+
+```
+Span.attributes > ScopeSpans.scope.attributes > ResourceSpans.resource.attributes
+```
+
+#### Span Events
+
+```
+Span.events.attributes > Span.attributes > ScopeSpans.scope.attributes > ResourceSpans.resource.attributes
+```
+
+#### Span links
+
+```
+Span.links.attributes > Span.attributes > ScopeSpans.scope.attributes > ResourceSpans.resource.attributes
```

Comment:
I'm a bit uncertain if link attributes should override the rest of attributes. What comes to mind is, a Link contain an attribute, say, `host.id` that was obtained from the span of a different trace, from a different host. If we do what's proposed here, we might override the `host.id` that is in the current span, potentially mixing up host ids. I'm not sure this is realistic or if it's just me, but I feel there's a problem hidden in here.

Reply:
Solved in other comments.